### PR TITLE
[FLOC-3799] Collect artifacts (logs) from cron-triggered acceptance test jobs

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://clusterhq.atlassian.net/browse/FLOC-3799

This removes the separate `copyArtifact`-calling loop over some of the jobs (all except the cron jobs) and replaces it with a `copyArtifact` call in the `build_steps` method that is called for every job.

This has two consequences:
  * The `CopyArtifact` configuration is now part of each job rather than all of the `CopyArtifact` configuration being part of the main multijob (incidental but, I think, an improvement).
  * Cron-style jobs now have `CopyArtifact` configuration if they define artifacts that should be copied (the actual point of this PR).

Visual inspection of the generated configuration xml suggest these are actually the consequences achieved by this PR.

### CODE REVIEW

File: jobs.groovy.j2 [hunk 1]
```diff
@@ -289,14 +307,36 @@ def build_publishers(v) {
...
             }
         }
+
+        /* not every job produces an artifact, so make sure we
+           don't try to fetch artifacts for jobs that don't
+           produce them */
+        if (job.archive_artifacts) {
+            for (artifact in job.archive_artifacts) {
+                copyArtifacts(
+                    "${dash_project}/${dash_branch_name}/${job_name}") {
```
Comment (james-w) [inline jobs.groovy.j2]: :question: This looks like it would be a common pattern, could it be a function?

**Comment by james-w** [comment]
Hi, thanks for the branch, it looks like a better organisation anyway. Special
thanks for better documenting how to test changes.

:+1: Please land as you like, pending mentor review.

@jml mentor review please.

**Comment by jml** [comment]
Mentor review approval. No feedback, meta or otherwise.

File: jobs.groovy.j2 (hunk 1 above)
Comment (exarkun) [inline jobs.groovy.j2]: Yea, I agree that would make sense.  I started down this road while working on this and encountered some unexpected difficulties that made me set that aside.  I may make another attempt if I get a chance, or we should at least suggest this to whoever does the next significant bit of work on this file. :)

**Comment by exarkun** [comment]
Thanks!  Merging now.

**Pull request merged**